setup.py

```diff
@@ -6,13 +8,32 @@ def readme():
         return f.read()
 
 
+def _is_old_setuptool():
+    """
+    Return True if setuptool is older than 20.5 which introduced environment
+    markers for requirements:
+        https://www.python.org/dev/peps/pep-0508/#environment-markers
+    """
+    major, minor = setuptools.__version__.split('.')[:2]
+    return (int(major), int(minor)) < (20, 5)
+
+
 def requirements():
     req = []
-    with open('requirements.txt') as f:
-        for l in f:
-            l.strip()
-            if not l.startswith('#'):
-                req.append(l)
+    flag_old = _is_old_setuptool()
```

Comment:
IMHO in general it is nicer to give such flag variables explicit names, for example `setuptools_outdated`. Nevertheless. I wonder if this flag is needed at all, just change the if conditions (first check for the environment marker, then call the function once) and it would be as efficient currently. Just want to give you full feedback. I am not sure if this code actually needs this much scrutiny.

Reply:
Changed to `setuptools_is_outdated = _is_outdated_setuptools()`